File: packages/elixir-sdk/lib/treedx_sdk/adapters/migrations.ex
defmodule TreeDxSdk.Migrations do
  @moduledoc false
  alias TreeDxSdk.Adapters.Common

  def create(client, repo_id, body) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/migrations",
      body,
      %{}
    )
  end

  def get(client, repo_id, migration_id) do
    Common.json_request(
      client,
      :get,
      "/api/v1/repos/" <>
        Common.segment(repo_id) <> "/migrations/" <> Common.segment(migration_id),
      nil,
      %{}
    )
  end
end

File: packages/rust-sdk/src/adapters/migrations.rs
use std::sync::Arc;

use serde_json::Value;

use crate::adapters::common::{json_request, segment};
use crate::error::TreeDxResult;
use crate::transport::{Transport, TreeDxHttpMethod};

#[derive(Clone)]
pub struct MigrationsAdapter {
    transport: Arc<dyn Transport>,
}

impl MigrationsAdapter {
    pub fn new(transport: Arc<dyn Transport>) -> Self {
        Self { transport }
    }

    pub async fn create(&self, repo_id: &str, body: Value) -> TreeDxResult<Value> {
        json_request(
            &self.transport,
            TreeDxHttpMethod::Post,
            format!("/api/v1/repos/{}/migrations", segment(repo_id)),
            Some(body),
            None,
        )
        .await
    }

    pub async fn get(&self, repo_id: &str, migration_id: &str) -> TreeDxResult<Value> {
        json_request(
            &self.transport,
            TreeDxHttpMethod::Get,
            format!(
                "/api/v1/repos/{}/migrations/{}",
                segment(repo_id),
                segment(migration_id)
            ),
            None,
            None,
        )
        .await
    }
}
